pkg/scheduler/core/extender.go

```diff
@@ -299,8 +301,14 @@ func (h *HTTPExtender) Filter(
 
 	if h.nodeCacheCapable && result.NodeNames != nil {
 		nodeResult = make([]*v1.Node, 0, len(*result.NodeNames))
-		for i := range *result.NodeNames {
-			nodeResult = append(nodeResult, nodeNameToInfo[(*result.NodeNames)[i]].Node())
+		for _, nodeName := range *result.NodeNames {
+			if node, ok := nodeNameToInfo[nodeName]; ok {
+				nodeResult = append(nodeResult, node.Node())
```

Comment:
With this PR, `len(nodeResult)` is not necessarily the same as `len(*result.NodeNames)`, so it'd good to change L303 to `nodeResult = make([]*v1.Node)`.

Reply:
NP, and thanks for your review. :)

For this comment, do you mean change to `nodeResult = make([]*v1.Node, 0)`, however, `len(nodeResult)` is **expected** to be the same as `len(*result.NodeNames)`, and in **common cases**, it is. Otherwise, it is a **rare** "exception", and we already returned an error for this.

So, **to optimize for most common cases**, we would better to still make a slice with an expect capacity for the following fast appending.

Is that ok?

Reply:
> For this comment, do you mean change to nodeResult = make([]*v1.Node, 0)

Yes, if the length is not fixed.

According to your comments that the length is fixed, and we return error upon mismatch. **_Technically_** we should do `nodeResult = make([]*v1.Node, len(*result.NodeNames))` to ensure `len` and `cap` inside the slice are the same, and use `nodeResult[i] = node.Node()`. (I recalled that I did a test and it showed a perf improvement).

Can you update the code, as well as L305 to L310, and make them squashed into one commit?

Reply:
Thanks! Adjusted according to your suggestion, as well as L311 to L314, pls check